File: Plugins/Aspose_Barcode_Java_for_PHP/src/aspose/barcode/WorkingWithBarcode/AdvanceBarcodeFeatures/UseChecksumAndSupplementData.php
<?php
namespace Aspose\Barcode\WorkingWithBarcode\AdvanceBarcodeFeatures;

use com\aspose\barcode\BarCodeBuilder as BarCodeBuilder;
use com\aspose\barcode\Symbology as Symbology;
use com\aspose\barcode\EnableChecksum as EnableChecksum;

class UseChecksumAndSupplementData{

    public static function run($dataDir=null){

        # Instantiate barcode object
        $bb = new BarCodeBuilder();

        # Set the Code text for the barcode
        $bb->setCodeText("1234567");

        # Set the symbology type to Code39
        $symbology=new Symbology();
        $bb->setSymbologyType($symbology->Code39Standard);

        # Make the checksum to be visible on the barcode
        $enableChecksum=new EnableChecksum();
        $bb->setEnableChecksum($enableChecksum->Default);

        # Save the image to disk in JPG format
        $bb->save($dataDir . "barcode.jpg");

        # Print message
        print "Barcode image generated successfully.".PHP_EOL;
    }
}

File: Plugins/Aspose.BarCode Java for Jython/asposebarcode/WorkingWithBarcode/AdvanceFeatures/UseChecksumAndSupplementData.py
from asposebarcode import Settings
from com.aspose.barcode import BarCodeBuilder
from com.aspose.barcode import Symbology
from com.aspose.barcode import EnableChecksum

class UseChecksumAndSupplementData:

    def __init__(self):
        dataDir = Settings.dataDir + 'WorkingWithBarcode/AdvanceBarcodeFeatures/UseChecksumAndSupplementData/'
        
        # Instantiate barcode object
        bb = BarCodeBuilder()

        # Set the Code text for the barcode
        bb.setCodeText("1234567")

        # Set the symbology type to Code39
        symbology=Symbology
        bb.setSymbologyType(symbology.Code39Standard)

        # Make the checksum to be visible on the barcode
        enableChecksum=EnableChecksum
        bb.setEnableChecksum(enableChecksum.Default)

        # Save the image to disk in JPG format
        bb.save(dataDir + "barcode.jpg")

        # Print message
        print "Barcode image generated successfully."

if __name__ == '__main__':        
    UseChecksumAndSupplementData()